Fix #5749: Add support for checking lesson compatibility weekly

BenHenning opened the pull request:
## Explanation
Fixes #5749

This PR introduces support for, once per week (Mondays at 2:30am GMT), attempting to download the list of lessons and failing if any conversion errors occur (support for this was introduced in the asset download script in 0252fc8fd5b246e56a3a34c88018eda2bf3a09f3 and 0bec9004cdb6faaa899d8b03162bce6b750d47de). There are currently over 6000 issues that need fixing, so it's expected that this is a cron job that will be failing for some time as the content team works through the failures, and the web team introduces checks to help prevent some of these classes of failures from regressing.

https://github.com/BenHenning/oppia-android/actions/runs/13843084590/job/38735087365 demonstrates a failure when issues are detected:

![image](https://github.com/user-attachments/assets/ced33349-f83e-40bf-9613-368fdb8c1c1f)

Note that the actual exception thrown is caught as a GitHub actions annotation, but fortunately it doesn't have any critically important information as all the actual errors are printed to standard output.

## Essential Checklist
- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only
N/A -- This is a CI infrastructure change.

BenHenning committed:
Create lesson_checks.yml

BenHenning committed:
Update lesson_checks.yml

Fix reference to PROD_SERVER_LESSON_SECRET.

BenHenning commented:
PTAL @adhiamboperes for code owners, and @seanlip for approval on approach.

Note that I haven't actually tested this--may try to later in a fork. :)

adhiamboperes reviewed (approved):
LGTM for codeowners!

seanlip commented on .github/workflows/lesson_checks.yml:
```diff
@@ -0,0 +1,68 @@
+jobs:
+  check_lesson_compatibility:
+    name: Check production lesson conversion
+    runs-on: ubuntu-20.04
```
Most Web actions are ubuntu-22.04 FYI -- this might become obsolete earlier as time goes on.

BenHenning replied:
Our actions can be much more sensitive to OS changes than web's. I don't mind testing Ubuntu 22, but we don't take latest generally until we start getting a push to upgrade.

BenHenning replied:
Updated to 22.04 and it seems to run fine, which is nice to know. That means there's a good chance the rest of the workflows could be safely updated in the future, as well.

Resolving since you approved and this is a trivial change that was completed.

seanlip commented on .github/workflows/lesson_checks.yml:
```diff
@@ -0,0 +1,68 @@
+name: Checks for Oppia Lesson Compatibility
+
+on:
+  workflow_dispatch:
```
This means it can be triggered manually, right? If so, that's great -- just wanted to make sure we have that functionality since I think there will be times where we'd want to generate a report on demand.

BenHenning replied:
Yup, that's exactly what this trigger is used for.

BenHenning replied:
Nothing more to do here, so resolving.

seanlip reviewed (approved):
Thanks @BenHenning! Looks good, let's go ahead and run it and see how it goes :)

BenHenning committed:
Update lesson_checks.yml

BenHenning committed:
Update lesson_checks.yml

Remove unnecessary extra pathing for checkout to simplify (and possibly fix) the later commands.

BenHenning committed:
Update lesson_checks.yml

Remove invalid comment and fix develop subpath reference.

BenHenning committed:
Update lesson_checks.yml

Generalize the API key step & ensure release_assets exists since the download_lesson_list script won't create it.

BenHenning reviewed (commented):
Latest changes are minor and were necessary to get the workflow to actually operate correctly. Seems reasonable to merge without a follow-up review.

Thanks @adhiamboperes & @seanlip!
